tasks/build.yml

```diff
@@ -20,6 +20,17 @@
     src="{{python_src_prefix}}/{{python_src_pkg}}"
     dest="{{python_src_prefix}}"
 
+- name: Patch code
```

Comment:
sha1 checksum checking for get_url is only there for ansible 2.0 and above. Prior versions support only the sha256sum argument, which is deprecated on 2.0, so maintaining cross compatibility becomes very cumbersome.

Idempotence is already broken since get_url in the previous step is currently unconditional (when is commented), unarchive is unconditional as well, and the build is currently conditional only on python not being installed (so it won't pick up changes in source code even if all the above tasks actually report changed or not correctly).

You can never fully ascertain which patches were applied also, there's no way to know whether a binary was built with or without a patch. So even if the patches were left on the source path, that still doesn't guarantee they were built into the binary.

The other changes don't seem like a big deal